python/audit/utils.py

```diff
@@ -0,0 +1,59 @@
+from functools import wraps
+
+from audit.models import (
+    AUDIT_OPERATION_MAP,
+    AuditLog,
+    AuditLogOperationResult,
+    AuditObject,
+)
+from cm.errors import AdcmEx
+from django.views.generic.base import View
+from rest_framework.status import is_success
+
+
+def audit(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        error = None
+
+        try:
+            resp = func(*args, **kwargs)
+            status_code = resp.status_code
+        except AdcmEx as exc:
+            error = exc
+            resp = None
+            status_code = exc.status_code
+
+        view: View = args[0]
+        audit_operation = AUDIT_OPERATION_MAP.get(view.__class__.__name__)
+        object_changes = {}
+
+        if is_success(status_code):
+            operation_result = AuditLogOperationResult.Success
+            if resp.data:
+                audit_object = AuditObject.objects.create(
+                    object_id=resp.data.serializer.instance.id,
+                    object_name=resp.data.serializer.instance.name,
```

Comment:
I think there won't be `name` for host entities, it'll be `fqdn`

Reply:
It'll be fixed after hosts audit implementation

Reply:
okay
we practice resolving comments by author